mzLib/Proteomics/Protein/Protein.cs

```diff
@@ -832,33 +834,75 @@ public static Protein ScrambleDecoyProteinSequence(
             }
 
             string scrambledProteinSequence = originalDecoyProtein.BaseSequence;
+            // Clone the original protein's modifications
+            var scrambledModificationDictionary = originalDecoyProtein.OriginalNonVariantModifications.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+
```

Comment:
Instead of cloning, then checking if anything is contained later, could you scramble, check if there are mods, then clone?